### ISSUE TITLE
Warning about unmounting encrypted partition not helpful (change unmount->eject)

### ISSUE THREAD

**Issue by alex-nitrokey** [Nitrokey/nitrokey-app#322]
Hi,
talking about this message:
https://github.com/Nitrokey/nitrokey-app/blob/f63040e66c7f1d1783fb001db99d65bac752433c/i18n/nitrokey_de_DE.ts#L1480

I understand, what is meant and many techy people will mostly do. But I am not sure, if this is really helpful for average users. Maybe we should delete:
> To avoid data loss, please unmount the partitions before proceeding.

I thought about this message, because a user told me that it is totally unclear to him, what is meant and because I often realized that users are frightened and uncertain because of this message. 
I think that it should be deleted at all because

* Windows did warn users in the past, when unplugging USB storage without prior safely unmounting it. They eventually stopped, I think because obviously nobody cared anyway...
* therefore: nowadays users are aware of the fact that unfinished tasks will break, if plugging out a pendrive and still will do if they like to do so
* on the other side, many users are not aware on what "safely unmounting" really is, so telling them to do so is not helping them in any way
* on the contrary: they got frightened they have to do anything although they probably don't. As many systems are just mounting devices automatically, most users will get this message although they did not change the storage at all. -> they will ignore the message eventually.

I totally see the point in warning the user here. If you decide to keeping it this way, we may think about a text which is more clear and may even give a link where it is shown how to do it?

As a proposal:
> [...] Do you want to proceed? Running operations are terminated.

**Comment by szszszsz** [comment]
Hi @alex-nitrokey !

It is great you are working on simplifying the UI - nitrokey-app seems to need it. 
I understand, that `unmounting` might not be widely known concept among Windows, however Linux and MacOS users should be accustomed to them. Rephrasing it is desired indeed (perhaps even per OS; with a link to a broader description, what unmounting/ejecting really means). However, I believe current message carries more information than your proposition (which is even more general and scares me more ;-) ).

I would not agree with skipping it at all - this is a real danger to the data. With the obscured writing by the OS (caching writing operations to memory and executing in the background, or even directly on unmount) user might not understand, why the data are corrupted and might blaming the device. This is especially an issue with lower writing speeds and modern OSes.

Perhaps we could somehow detect is the device under writing operation (e.g. by measuring the time from last write to the encrypted/hidden volumes) and then show mentioned warning message (but that only fixes part of the problem).
Nevertheless, I believe it would be better instead to ask user to read brief quick-start guide about App and Storage usage, so the warning would be a reminder and not a confusing, scary message.

A guide, which would focus exactly how to use, lock, and recover the device would be perfect. Covering and explaining possible cases would make users more convinced about their actions. Example topics:
- What would happen if I remove the device with encrypted volume unlocked?
- What happens if I enter the PIN 3 times wrong?
- What factory reset does?
- What if I will not synchronize the time on the device?

Even better perhaps, if the dialog's would be enhanced with `?` mark describing the possible outcomes of given action. 'First-use wizard' might help as well, detailing the usage by the occasion of running each operation.

Let me know what you think.

**Comment by jans23** [comment]
Perhaps we should try to find a better wording. @alex-nitrokey Do you know the terminology which has been used by Windows? I think in German they used "auswerfen" which would translate to "eject". Would that be less confusing than "unmounting"?

**Comment by szszszsz** [comment]
Bump

**Comment by alex-nitrokey** [comment]
> Nevertheless, I believe it would be better instead to ask user to read brief quick-start guide about App and Storage usage, so the warning would be a reminder and not a confusing, scary message.
A guide, which would focus exactly how to use, lock, and recover the device would be perfect. Covering and explaining possible cases would make users more convinced about their actions. Example topics:
Even better perhaps, if the dialog's would be enhanced with `?` mark describing the possible outcomes of given action. 'First-use wizard' might help as well, detailing the usage by the occasion of running each operation.

Well, I think you are right, that's probably best. Thinking about it the problem is not so much the message itself, but the fact that users don't know, what it means. Having such detailed instructions and explanations should helping alot. We should probably do them anyway. So link to the website is probably best? I like the question mark idea but I am not sure if this can take enough information?

I want to do some basic step-by-step guides in the next... months(?) anyway. This should one of them. I can not say when it will be ready. It's a good idea to keep this open until then...

> Do you know the terminology which has been used by Windows? I think in German they used "auswerfen" which would translate to "eject". Would that be less confusing than "unmounting"?

I don't think that this would help much. For the user it is pluging in and out, they don't car or know what the system is really doing. Especially in a environment in which devices will be automounted anyway.

**Comment by szszszsz** [comment]
Yes, the information we could insert to help boxes is limited. It should briefly describe the terms, but also redirect for more to the proper guide or the main site.
Should I close this then as non-issue? Could you summarize, what is applicable to the App from our discussion?

**Comment by jans23** [comment]
As a quick fix I suggest to change the wording from "unmount"/"aushängen" to "eject"/"auswerfen". This may not be sufficient but I think a bit more common terminology.

**Comment by szszszsz** [comment]
@alex-nitrokey Could you make a PR with Jan's proposition?

**Comment by alex-nitrokey** [comment]
Sure!
